File: assets/codebase/menu/MenuOverlayImageDefinition.cs
namespace city.menu {
    /// <summary>
    /// Describes one decorative city menu overlay image baked into the generated demo-disc scene.
    /// </summary>
    public sealed class MenuOverlayImageDefinition {
        /// <summary>
        /// Initializes one decorative city menu overlay image definition.
        /// </summary>
        /// <param name="texturePath">Project-relative texture source path.</param>
        /// <param name="width">Authored overlay width in canvas pixels.</param>
        /// <param name="height">Authored overlay height in canvas pixels.</param>
        /// <param name="bottomMargin">Bottom margin from the reference canvas edge in pixels.</param>
        /// <param name="rightMargin">Right margin from the reference canvas edge in pixels.</param>
        public MenuOverlayImageDefinition(string texturePath, int width, int height, int bottomMargin, int rightMargin) {
            if (string.IsNullOrWhiteSpace(texturePath)) {
                throw new ArgumentException("Texture path must be provided.", nameof(texturePath));
            }
            if (width < 1) {
                throw new ArgumentOutOfRangeException(nameof(width), "Overlay width must be positive.");
            }
            if (height < 1) {
                throw new ArgumentOutOfRangeException(nameof(height), "Overlay height must be positive.");
            }
            if (bottomMargin < 0) {
                throw new ArgumentOutOfRangeException(nameof(bottomMargin), "Bottom margin must not be negative.");
            }
            if (rightMargin < 0) {
                throw new ArgumentOutOfRangeException(nameof(rightMargin), "Right margin must not be negative.");
            }

            TexturePath = texturePath;
            Width = width;
            Height = height;
            BottomMargin = bottomMargin;
            RightMargin = rightMargin;
        }

        /// <summary>
        /// Gets the project-relative texture source path.
        /// </summary>
        public string TexturePath { get; }

        /// <summary>
        /// Gets the authored overlay width in canvas pixels.
        /// </summary>
        public int Width { get; }

        /// <summary>
        /// Gets the authored overlay height in canvas pixels.
        /// </summary>
        public int Height { get; }

        /// <summary>
        /// Gets the authored bottom margin from the reference canvas edge in pixels.
        /// </summary>
        public int BottomMargin { get; }

        /// <summary>
        /// Gets the authored right margin from the reference canvas edge in pixels.
        /// </summary>
        public int RightMargin { get; }
    }
}

File: windows-build-20260602-cube-test-ground-high-static-camera/cooked/code/gameplay/MenuOverlayImageDefinition.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "MenuOverlayImageDefinition.hpp"
#include "runtime/native_string.hpp"
#include "runtime/native_exceptions.hpp"
#include "MenuOverlayImageDefinition.hpp"
#include "runtime/native_exceptions.hpp"
#include "runtime/native_string.hpp"

int32_t MenuOverlayImageDefinition::get_BottomMargin()
{
return this->BottomMargin;
}

int32_t MenuOverlayImageDefinition::get_Height()
{
return this->Height;
}

int32_t MenuOverlayImageDefinition::get_RightMargin()
{
return this->RightMargin;
}

const std::string& MenuOverlayImageDefinition::get_TexturePath()
{
return this->TexturePath;
}

int32_t MenuOverlayImageDefinition::get_Width()
{
return this->Width;
}

MenuOverlayImageDefinition::MenuOverlayImageDefinition(std::string texturePath, int32_t width, int32_t height, int32_t bottomMargin, int32_t rightMargin) : BottomMargin(0), Height(0), RightMargin(0), TexturePath(), Width(0)
{
    if (String::IsNullOrWhiteSpace(texturePath))
    {
throw ([&]() {
auto __ctor_arg_0000003E = "Texture path must be provided.";
auto __ctor_arg_0000003F = "texturePath";
return new ArgumentException(__ctor_arg_0000003E, __ctor_arg_0000003F);
})();
    }
    if (width < 1)
    {
throw ([&]() {
auto __ctor_arg_00000040 = "width";
auto __ctor_arg_00000041 = "Overlay width must be positive.";
return new ArgumentOutOfRangeException(__ctor_arg_00000040, __ctor_arg_00000041);
})();
    }
    if (height < 1)
    {
throw ([&]() {
auto __ctor_arg_00000042 = "height";
auto __ctor_arg_00000043 = "Overlay height must be positive.";
return new ArgumentOutOfRangeException(__ctor_arg_00000042, __ctor_arg_00000043);
})();
    }
    if (bottomMargin < 0)
    {
throw ([&]() {
auto __ctor_arg_00000044 = "bottomMargin";
auto __ctor_arg_00000045 = "Bottom margin must not be negative.";
return new ArgumentOutOfRangeException(__ctor_arg_00000044, __ctor_arg_00000045);
})();
    }
    if (rightMargin < 0)
    {
throw ([&]() {
auto __ctor_arg_00000046 = "rightMargin";
auto __ctor_arg_00000047 = "Right margin must not be negative.";
return new ArgumentOutOfRangeException(__ctor_arg_00000046, __ctor_arg_00000047);
})();
    }
this->TexturePath = texturePath;
this->Width = width;
this->Height = height;
this->BottomMargin = bottomMargin;
this->RightMargin = rightMargin;
}

